zebra-rpc/src/methods.rs

```diff
@@ -315,3 +353,9 @@ pub struct GetBlock(#[serde(with = "hex")] SerializedBlock);
 #[derive(Debug, PartialEq, serde::Serialize)]
 /// Response to a `getbestblockhash` RPC request.
 pub struct GetBestBlockHash(#[serde(with = "hex")] block::Hash);
+
+#[derive(Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+/// Response to a `getrawmempool` RPC request.
+///
+/// A JSON array of string with the transaction hashes.
+pub struct GetRawMempool(HashSet<transaction::Hash>);
```

Comment:
I am unsure if we should have `#[serde(with = "hex")]` here and if we do how to implement tohex and fromhex for a hashset.

Reply:
I think you don't need this type. You'll have to return a `Vec<_>` from the RPC method so that it can be serialized as a JSON array.

As for the items, I think you'll have to convert from the `HashSet<transaction::Hash>` collection into a `Vec<String>` with the hexadecimal strings of the transaction hashes. I'd suggest something like:

```rust
unmined_transaction_ids
    .iter()
    .map(|id| id.mined_id().encode_hex())
    .collect()
```

Reply:
Works great, thanks!

Reply:
I forgot to remove the `GetRawMempool` type, now that it's no longer necessary :sweat_smile:

Reply:
Removed in https://github.com/ZcashFoundation/zebra/pull/3851/commits/25bdc0428d7060fca8e7c729cc02c57766e4760e and changed the docs as well.